tianshou/utils/moving_average.py

```diff
@@ -26,36 +27,38 @@ class MovAvg(object):
     def __init__(self, size: int = 100) -> None:
         super().__init__()
         self.size = size
-        self.cache = []
+        self.cache: List[Union[Number, np.number]] = []
         self.banned = [np.inf, np.nan, -np.inf]
 
-    def add(self, x: Union[float, list, np.ndarray, torch.Tensor]) -> float:
+    def add(
+        self, x: Union[Number, np.number, list, np.ndarray, torch.Tensor]
+    ) -> np.number:
         """Add a scalar into :class:`MovAvg`. You can add ``torch.Tensor`` with
         only one element, a python scalar, or a list of python scalar.
         """
         if isinstance(x, torch.Tensor):
             x = to_numpy(x.flatten())
         if isinstance(x, list) or isinstance(x, np.ndarray):
-            for _ in x:
-                if _ not in self.banned:
-                    self.cache.append(_)
-        elif x not in self.banned:
+            for i in x:
+                if i not in self.banned:
+                    self.cache.append(i)
+        elif np.isscalar(x) and x not in self.banned:
```

Comment:
`if isinstance(x, list) or isinstance(x, np.ndarray)` is not already doing the job ?

Reply:
Yes, you are right. I was previously in a mypy error and patch this fix, but now I find that removing this can also work well!